### DISCUSSION TITLE
SMI Counter in polling mode

### DISCUSSION THREAD

**Discussion by sensor-freak** [gemu2015/Sonoff-Tasmota#43] [General]
Hi,

I just realized that you've made quite some changes in the SMI driver. One of the changes has has some impact on the SMI Counter in polling mode.

The current code (in branch `development`) requires 6 consecutive readings of `low` after a trailing edge in order to identify a state change. Since the minimum polling interval seems to be something as low as 50ms, this causes a low phase of the input signal of about 300ms in order to be detected as valid state change. This might be much to slow, e.g. for water counters that generate 100 or 1000 (or even 10000) pulses per cubic meter.

Can you please explain why you did it this way? Did you have a problem with the previous debouncing method?

Kind regards

**Comment by gemu2015** [comment]
There was a major release in sml driver
however I did not change anything in counter handling for a very long time except these few lines from yesterday which doesn’t affect polling at all

**Comment by sensor-freak** [comment]
Aah, sorry. My fault. I didn't notice that the changes where only whitespace changes.

I'll move this discussion to the project.
